spec/structured-data.html

```diff
@@ -0,0 +1,111 @@
+      <p>The optional _reviver_ parameter is a function that takes two parameters, _key_ and _value_. It can filter and transform the results. It is called with each of the _key_/_value_ pairs produced by the parse, and its return value is used instead of the original value. If it returns what it received, the structure is not modified. If it returns *undefined* then the property is deleted from the result.</p>
+      <emu-alg>
+        1. Let _unfiltered_ be ? ParseJSONText(_text_).
+        1. Return ? BuildImmutableProperty(_unfiltered_, *""*, _reviver_).
+      </emu-alg>
+      <p>This function is the <dfn>%JSONParseImmutable%</dfn> intrinsic object.</p>
+      <p>The *"length"* property of the `parseImmutable` function is 2.</p>
+  
+      <emu-clause id="sec-buildimmutableproperty" aoid="BuildImmutableProperty">
+        <h1><ins>Runtime Semantics: BuildImmutableProperty ( _value_, _name_, _reviver_ )</ins></h1>
+        <p>The abstract operation BuildImmutableProperty takes arguments _value_ (a String, Number, Boolean, Null, or an Object), _name_ (a String), and _reviver_ (an optional function object). It performs the following steps when called:</p>
+        <emu-alg>
+          1. If Type(_value_) is Object, then
+            1. Let _isArray_ be ? IsArray(_val_).
+            1. If _isArray_ is *true*, then
+              1. Let _items_ be a new empty List.
+              1. Let _I_ be 0.
+              1. Let _len_ be ? LengthOfArrayLike(_val_).
+              1. Repeat, while _I_ &lt; _len_,
+                1. Let _childName_ be ! ToString(_I_).
+                1. Let _childValue_ be ? Get(_value_, ! _childName_).
+                1. Let _newElement_ be ? BuildImmutableProperty(_childValue_, _childName_, _reviver_).
+                1. Assert: Type(_newElement_) is not Object.
+                1. Append _newElement_ to _items_.
+                1. Set _I_ to _I_ + 1.
+              1. Let _immutable_ be a new Tuple value whose [[Sequence]] is _items_.
+            1. Else,
+              1. Let _props_ be ? EnumerableOwnPropertyNames(_obj_, ~key+value~).
+              1. Let _fields_ be a new empty List.
+              1. For each element _prop_ of _props_,
+                1. Let _childName_ be ! GetV(_prop_, `"0"`).
+                1. Let _childValue_ be ! GetV(_prop_, `"1"`).
+                1. Let _newElement_ be ? BuildImmutableProperty(_childValue_, _childName_, _reviver_).
+                1. Assert: Type(_newElement_) is not Object.
+                1. If _newElement_ is not *undefined*, then:
+                  1. Let _field_ be the Record { [[Key]]: _childName_, [[Value]]: _newElement_ }.
+                  1. Append _field_ to _fields_.
+              1. Let _immutable_ be a new Record value whose [[Fields]] is _fields_.
```

Comment:
Hmm, the fields here might be unsorted. We should probably make an abstract operation for deduplicating, sorting and making a Record. I think actually there used to be an operation for this, but then I told @rricard to inline it, but that was a bad suggestion :) This is also needed for Record.fromEntries and Record(), which also fail to sort the keys. (I guess just record literals and `fromEntries` actually need the deduplication, but it doesn't hurt to include that path for all sources of creating new Records.)